src/Base/BaseClass.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace BaseClass;
+
+/**
+ * Class Base
+ */
+
+abstract class BaseClass
```

Comment:
I consider the name "BaseClass" is not good. It can describe any kind of objects in general.
Please use other, more concrete name, like BaseProduct, BasePhone, BaseDevice (choose which describes your objects best)